scripts/upgrade-assure/README.md

```diff
@@ -1,75 +1,132 @@
+## Upgrade Steps
 
-```
-make install
-go run ./scripts/upgrade-assure/... https://snapshots-testnet.stake-town.com/elys/elystestnet-1_latest.tar.lz4 ~/go/bin/elysd ~/go/bin/elysd --skip-proposal
+### Step 1: Checkout the Implementation Branch
+
+Switch to the branch containing the new implementation:
+
+```bash
+git checkout estaking_impl
 ```
 
-## Polkachu testnet snapshots
+### Step 2: Create a New Tag and Install
 
-You can find the latest Polkachu testnet snapshot [here](https://polkachu.com/testnets/elys/snapshots).
+Tag the new release and install it:
 
-```
+```bash
+git tag v0.31.0
 make install
-go run ./scripts/upgrade-assure/... https://snapshots.polkachu.com/testnet-snapshots/elys/elys_5724942.tar.lz4 ~/go/bin/elysd ~/go/bin/elysd --skip-proposal
 ```
 
-## AviaOne testnet snapshots
+### Step 3: Retrieve the Current Binary for Darwin Users
 
-You can find the latest AviaOne testnet snapshot [here](https://aviaone.com/blockchains-service/elystestnet-1-elys.html#8).
+Download the specific binary for Darwin ARM64 architecture using `curl`:
 
-```
-make install
-go run ./scripts/upgrade-assure/... https://services.elystestnet-1.elys.aviaone.com/elystestnet-1_2024-03-06.tar.gz ~/go/bin/elysd ~/go/bin/elysd --skip-proposal
+```bash
+curl -L https://github.com/elys-network/elys/releases/download/v0.29.31/elysd-v0.29.31-darwin-arm64 -o /tmp/elysd-v0.29.31
 ```
 
-# How can I perform a test with a version upgrade that involves extensive changes to data structures?
+### Step 4: Retrieve Testnet Snapshot
 
-```
-git checkout v0.29.30
-make install
-cp -a ~/go/bin/elysd /tmp/elysd-v0.29.30
-```
+Fetch the latest testnet snapshot, necessary for data migration using `curl`:
 
-```
-go run ./scripts/upgrade-assure/... --home /tmp/elys --home2 /tmp/elysd2 https://tools.highstakes.ch/files/elys.tar.gz /tmp/elysd-v0.29.30 /tmp/elysd-v0.29.31 --skip-node-start
+```bash
+curl -L https://snapshots-testnet.stake-town.com/elys/elystestnet-1_latest.tar.lz4 -o /tmp/snapshot.tar.lz4
 ```
 
-```
+### Step 5: Checkout the Previous Version
+
+Switch to the previously stable version:
+
+```bash
 git checkout v0.29.31
-make install
-cp -a ~/go/bin/elysd /tmp/elysd-v0.29.31
 ```
 
+### Step 6: Run Upgrade-Assure Script
+
+#### 6a: Initial Run
+
+Run the upgrade-assure script without starting the node:
+
+```bash
+go run ./scripts/upgrade-assure/... /tmp/snapshot.tar.lz4 /tmp/elysd-v0.29.31 ~/go/bin/elysd --skip-node-start
 ```
-go run ./scripts/upgrade-assure/... --home /tmp/elys --home2 /tmp/elysd2 https://tools.highstakes.ch/files/elys.tar.gz /tmp/elysd-v0.29.30 /tmp/elysd-v0.29.31 --skip-snapshot --skip-chain-init
-```
-# Troubleshooting
 
-These are some problems and its solutions when the script is executed
+#### 6b: Handle Potential Errors
+
+Address any type errors, such as difficulties in unmarshaling strings into integer fields in Go struct fields.
+
+#### 6c: Update the Script
```

Comment:
Is 6c step intended to fix errors faced on **6b** right? It would be nice you have a couple of examples here as well on what are the data structures that we need to keep an eye on, what's the path for those and what's the process to update them. Is this related to proto files? Is it an automated process?

Thank you!

Reply:
this will depends on what the error will be returned at the export step, I dont have an example right now but I would be happy to add one in subsequent updates when facing a new instance